core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -234,10 +234,25 @@ public boolean isLaunchSupported() {
         return launcher.isLaunchSupported();
     }
 
+    /**
+     * Return the {@code ComputerLauncher} for this SlaveComputer.
+     */
     public ComputerLauncher getLauncher() {
         return launcher;
     }
 
+    /**
+     * Return the {@code ComputerLauncher} for this SlaveComputer, strips off
+     * any {@code DelegatingComputerLauncher}s.
+     */
+    public ComputerLauncher getDelegatedLauncher() {
+        ComputerLauncher l = launcher;
+        while (l instanceof DelegatingComputerLauncher) {
+            l = ((DelegatingComputerLauncher)l).getLauncher();
```

Comment:
What if it the nested launcher is a delegating launcher as well?
I would expect the method to continue the recursion then.

Reply:
That's why it's while loop ;).

Reply:
Also included ComputerLauncherFilter which has the same problem.

Reply:
Sorry, probably I was blind :(